src/_c_internal_utils.c

```diff
@@ -1,11 +1,61 @@
 #define PY_SSIZE_T_CLEAN
 #include <Python.h>
+#ifdef __linux__
+#include <dlfcn.h>
+#endif
 #ifdef _WIN32
 #include <Objbase.h>
 #include <Shobjidl.h>
 #include <Windows.h>
 #endif
 
+static PyObject* mpl_display_is_invalid(PyObject* module)
+{
+#ifdef __linux__
+    void* libX11;
+    if ((libX11 = dlopen("libX11.so", RTLD_LAZY))) {
+        struct Display* display = NULL;
+        struct Display* (* XOpenDisplay)(char const*) =
+            dlsym(libX11, "XOpenDisplay");
+        int (* XCloseDisplay)(struct Display*) =
+            dlsym(libX11, "XCloseDisplay");
+        if (XOpenDisplay && XCloseDisplay
+                && (display = XOpenDisplay(NULL))) {
+            XCloseDisplay(display);
+        }
+        if (dlclose(libX11)) {
+            PyErr_SetString(PyExc_RuntimeError, dlerror());
+            return NULL;
+        }
+        if (display) {
+            Py_RETURN_FALSE;
+        }
+    }
+    void* libwayland_client;
+    if ((libwayland_client = dlopen("libwayland-client.so", RTLD_LAZY))) {
+        struct wl_display* display = NULL;
+        struct wl_display* (* wl_display_connect)(char const*) =
+            dlsym(libwayland_client, "wl_display_connect");
+        int (* wl_display_disconnect)(struct wl_display*) =
+            dlsym(libwayland_client, "wl_display_disconnect");
+        if (wl_display_connect && wl_display_connect
```

Comment:
Should this be `wl_display_connect && wl_display_disconnect`?

Reply:
yup, thanks